test/stripe/subscription_test.rb

```diff
@@ -63,9 +63,9 @@ class SubscriptionTest < Test::Unit::TestCase
     context "#delete_discount" do
       should "be able to delete a subscriptions's discount" do
         subscription = Stripe::Subscription.retrieve("sub_123")
-        subscription = subscription.delete_discount
+        discount = subscription.delete_discount
         assert_requested :delete, "#{Stripe.api_base}/v1/subscriptions/sub_123/discount"
-        assert subscription.is_a?(Stripe::Subscription)
+        assert discount.is_a?(Stripe::Discount)
```

Comment:
The old `delete_discount` method was actually bugged: it would return a `Stripe::Subscription` instance (because it used `initialize_from`) initialized with the payload of the deleted discount object.